Dynatrace/dynatrace-operator
# Remove semver validator for custom OneAgent version

## Description

[DAQ-1817](https://dt-rnd.atlassian.net/browse/DAQ-1817)

Custom OneAgent version on the DynaKube spec (e.g. `spec.oneAgent.cloudNativeFullStack.version`) doesn't work at the moment. We validate the CR expecting a semver (1.0.0) but the OA DaemonSet fails to start with a semver (with error: version malformed).

The quick fix for it is to remove the semver validator for the CR and then the field is usable again, only with timestamped versions, like 1.200.0.20240101-000000.

## How can this be tested?

- Apply a DynaKube with the following version, everything should work fine:
```yaml
spec:
  oneAgent:
    cloudNativeFullStack:
      version: 1.304.0.20240826-070516
```


[DAQ-1817]: https://dt-rnd.atlassian.net/browse/DAQ-1817?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
